lib/events/events.go

```diff
@@ -351,3 +353,47 @@ func Error(err error) *string {
 	str := err.Error()
 	return &str
 }
+
+// Wait for the given event to become available, or for a timeout, whichever is sooner.
+// Returns true if the event became available, or false if we timed out.
+func (s *bufferedSubscription) waitForNewEventWithTimeout(id int, timeout time.Duration) bool {
+	// Early exit, in the case where events are already available
+	s.mut.Lock()
+	cur := s.cur
+	s.mut.Unlock()
+	if id < cur {
+		return true
+	}
+
+	// From the point on, we know we're going to have to wait
+
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
```

Comment:
If hundreds of events are being created a second, then most of the polls should result in events already being available, in which case we don't create the timer. I'll leave as-is (and add a comment) though